HousingFinanceInterimApi/V1/Infrastructure/DatabaseContext.cs

```diff
@@ -410,6 +418,271 @@ await using (var reader = await command.ExecuteReaderAsync().ConfigureAwait(fals
+                {
+                    var result = new string[reader.FieldCount];
+                    for (int i = 0; i < reader.FieldCount; i++)
+                    {
+                        result[i] = reader[i].ToString();
+                    }
+
+                    results.Add(result);
+                }
+
+                results.Insert(0, columnNames);
+            }
+            dbConnection.Close();
+
+            return results;
+        }
+
+        public async Task<List<string[]>> GetCashImportByDateAsync(DateTime startDate, DateTime endDate)
+        {
+            var results = new List<string[]>();
+
+            var dbConnection = Database.GetDbConnection() as SqlConnection;
+            var command = new SqlCommand($"usp_GetCashImportByDate", dbConnection)
+            {
+                CommandTimeout = 900,
+                CommandType = CommandType.StoredProcedure
+            };
+
+            command.Parameters.AddWithValue("@startDate", startDate.ToString("yyyy-MM-dd"));
+            command.Parameters.AddWithValue("@endDate", endDate.ToString("yyyy-MM-dd"));
+
+            dbConnection.Open();
+            await using (var reader = await command.ExecuteReaderAsync().ConfigureAwait(false))
+            {
+                var columnNames = new string[reader.FieldCount];
+                for (int i = 0; i < reader.FieldCount; i++)
+                {
+                    columnNames[i] = reader.GetName(i);
+                }
+
+                while (reader.Read())
+                {
+                    var result = new string[reader.FieldCount];
+                    for (int i = 0; i < reader.FieldCount; i++)
+                    {
+                        result[i] = reader[i].ToString();
+                    }
+
+                    results.Add(result);
+                }
+
+                results.Insert(0, columnNames);
+            }
+            dbConnection.Close();
+
+            return results;
+        }
+
+        public async Task<List<string[]>> GetChargesByYearAndRentGroupAsync(int year, string rentGroup)
+        {
+            var results = new List<string[]>();
+
+            var dbConnection = Database.GetDbConnection() as SqlConnection;
+            var command = new SqlCommand($"dbo.usp_GetChargesByYearAndRentGroup {year}, {rentGroup}", dbConnection)
+            {
+                CommandTimeout = 900
+            };
+
+            dbConnection.Open();
+            await using (var reader = await command.ExecuteReaderAsync().ConfigureAwait(false))
+            {
+                var columnNames = new string[reader.FieldCount];
+                for (int i = 0; i < reader.FieldCount; i++)
+                {
+                    columnNames[i] = reader.GetName(i);
+                }
+
+                while (reader.Read())
+                {
+                    var result = new string[reader.FieldCount];
+                    for (int i = 0; i < reader.FieldCount; i++)
```

Comment:
I don't think you need another for-loop inside a while (reader.Read(())
You could do 
`var result = reader.Select(r=>r["FieldName"].ToString()).ToArray();`